githubapp-token/action.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright 2022 Chainguard, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+name: Authenticate as a GitHub App
+description: |
+  This action generates a GitHub App token for a specific app and
+  installation ID. It requires having access to the app's private
+  key.
+
+inputs:
+  app_id:
+    description: |
+      The App ID of the GitHub App
+    required: true
+
+  installation_id:
+    description: |
+      The installation against which to authenticate.
+    required: true
+
+  private_key:
```

Comment:
do you want to use a secret for this?
the difference is as a secret, it will not show up in the logs.

https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow